### PR TITLE
Add bra-ket notation

### PR DESCRIPTION
**Pull request by ruiokada**
Hello,

I would like support for some bra-ket notation macros. I've tested the macros on KaTeX.org and the following screenshot is the result I got:

Screen Shot 2019-12-01 at 12 27 08 PM

Please let me know if there is anything else that needs to be done. Thanks!

### CODE REVIEW

**Comment by ruiokada** [comment]
[clabot:check]

File: src/macros.js [hunk 1]
```diff
@@ -960,6 +960,15 @@ defineMacro("\\argmin", "\\DOTSB\\operatorname*{arg\\,min}");
...
 defineMacro("\\plim", "\\DOTSB\\mathop{\\operatorname{plim}}\\limits");
 
+//////////////////////////////////////////////////////////////////////
+// braket.sty
+// http://ctan.math.washington.edu/tex-archive/macros/latex/contrib/braket/braket.pdf
+
+defineMacro("\\bra", "\\mathinner{\\langle{#1}|}");
+defineMacro("\\ket", "\\mathinner{|{#1}\\rangle}");
+defineMacro("\\braket", "\\mathinner{\\langle{#1}\\rangle}");
+defineMacro("\\ketbra", "\\mathinner{|{#1}\\rangle\\langle{#2}|}");
```
Comment (ylemkimon) [inline src/macros.js]: It seems `braket` doesn't define `\ketbra`:
> NOT defined is "\ketbra" (for projection operators) because I prefer \ket{ } \bra{ }.

Comment (ruiokada) [reply src/macros.js]: Hi, I've removed `\ketbra` and added `\Bra` and `\Ket` to be more faithful to the original `braket`.

**Comment by ronkok** [comment]
Good stuff.

Please also update the documentation file at `./docs/supported.md`.

Another help would be a test in `./test/katex-spec.js` to ensure that this builds.

**Comment by ruiokada** [comment]
Sorry for the billion commits, a lot unexpected things happened with the docs. I added everything to `./docs/supported.md` under a new section called "Special Notation" since I don't think bra-kets qualify as explicit functions or operators. I added tests for bra-kets in `./test/katex-spec.js` by following suit for the statmath package and adding tests for what `\ket{\psi}` etc. should parse as.

**Comment by kevinbarabash** [comment]
@ruiokada don't worry about the extra commits, we squash merge all PRs.  Thanks for updating the docs and adding tests.

File: src/macros.js [hunk 2]
```diff
@@ -930,6 +930,16 @@ defineMacro("\\argmin", "\\DOTSB\\operatorname*{arg\\,min}");
 defineMacro("\\argmax", "\\DOTSB\\operatorname*{arg\\,max}");
 defineMacro("\\plim", "\\DOTSB\\mathop{\\operatorname{plim}}\\limits");
 
+//////////////////////////////////////////////////////////////////////
+// braket.sty
+// http://ctan.math.washington.edu/tex-archive/macros/latex/contrib/braket/braket.pdf
```
Comment (kevinbarabash) [inline src/macros.js]: Thanks for reference the ctan package and linking to its docs.

**Comment by spino17** [comment]
Can anyone please tell me how to use this with var html = katex.renderToString("c = \\ket{\\Phi}", {
    throwOnError: false
}); 
I am not able to get bra-ket notation rendered.

**Comment by Yue-Zhengyuan** [comment]
Is there any plan to provide support for `\Braket` (with **uppercase B**)? Current `\braket` does not adjust its size with the contents.

**Comment by kevinbarabash** [comment]
@Yue-Zhengyuan you could probably define `\Bracket` using a macro.  I'm not sure how `\Bracket` adjusts the size of its contents though.

**Comment by ruiokada** [comment]
@kevinbarabash The `\Braket` implementation in braket resizes content by using`\right` + `\left` (to resize the angle brackets) and what seems like some search/replace function (to resize pipes (`|`) and `\vert`s by prepending them with `\middle`). The `\right` + `\left` part can be easily replicated in KaTeX with the macro `\gdef\Braket#1{\left\langle{#1}\right\rangle}`. The search and replace will take a bit more work within KaTeX itself but I'm not familiar enough with KaTeX's codebase to see how readily this can be implemented.

**Comment by edemaine** [comment]
Note that there's already a prototype implementation in #3171.

**Pull request merged**